src/tools/utils/index.tsx

```diff
@@ -69,17 +70,32 @@ const statuses = Object.values(Status);
 export const getStatusFromResponse = async (
   jobType: JobTypes,
   response: Response
-): Promise<[status: Status, idMappingResultsUrl?: string]> => {
+): Promise<
+  [status: Status, progress?: number, idMappingResultsUrl?: string]
+> => {
   let status: Status | undefined;
+  let progress: number | undefined;
   let idMappingResultsUrl: string | undefined;
   switch (jobType) {
     case JobTypes.ALIGN:
     case JobTypes.BLAST:
       status = (await response.text()) as Status;
       break;
-    case JobTypes.ASYNC_DOWNLOAD:
-      status = (await response.json()).jobStatus;
+    case JobTypes.ASYNC_DOWNLOAD: {
+      const data: ServerStatus = await response.json();
+      status = data.jobStatus;
+      if (data.totalEntries && data.processedEntries) {
+        // Round down to the closest integer
+        const potentialProgress = Math.floor(
+          // Make sure it never goes over 100%, just in case...
+          Math.min((data.processedEntries / data.totalEntries) * 100, 100)
```

Comment:
0 is fine because it falsy and will be ignored by the UI logic when deciding to render the retrieving vs writing to file message.
100% when flooring should only happen when the job is finished, so I guess that's fine.
The min(value, 100) is meant to not go over it as in the past in some tests the total was less than processed, but that was a bug, that makes me think that it might be better to display 120% and have people complain rather than 100% during 6 hours and people not realising it's actually doing more work and there's a bug that needs reporting.

What do you think of removing that Math.min call and just keeping the Math.floor ?

Reply:
I think removing `Math.min` could be a good strategy. We could also `console.error` in the case that the API starts giving >100% and pick this up in sentry.